app/models/manageiq/providers/vmware/infra_manager/event_catcher/runner.rb

```diff
@@ -42,27 +42,54 @@ def monitor_events
     reset_event_monitor_handle
   end
 
-  def process_event(event)
+  def event_dedup_key(event)
+    # duplicate the event but remove ids and timestamps that change with every event
+    # the remaining attributes are used to determine whether an event is a duplicate of another
+    event.except("key", "chainId", "createdTime").tap do |hash|
+      hash["info"] &&= hash["info"].except("key", "task", "queueTime", "eventChainId")
+    end
+
+    hash = event.except("key", "chainId", "createdTime")
```

Comment:
isnt this the first time `hash` is defined? 
I think the `event.tap` above is unused. But I might miss something here.